R/summarise_draws.R

```diff
@@ -124,23 +124,51 @@ summarise_draws.draws <- function(x, ..., .args = list()) {
   x <- repair_draws(x)
   x <- as_draws_array(x)
   variables <- variables(x)
-  out <- named_list(variables, values = list(named_list(names(funs))))
-  for (v in variables) {
-    draws <- drop_dims_or_classes(x[, , v], dims = 3, reset_class = TRUE)
-    args <- c(list(draws), .args)
-    for (m in names(funs)) {
-      out[[v]][[m]] <- do.call(funs[[m]], args)
-      if (rlang::is_named(out[[v]][[m]])) {
-        # use returned names to label columns
-        out[[v]][[m]] <- rbind(out[[v]][[m]])
-      }
+
+  # get length and output names, calculated on the first variable
+  v1 <- variables[1]
```

Comment:
do we know at this time that we have at least one variable in the draws object?

Reply:
We don't. Current behavior returns a 0x1 tibble in this case.  Shall I just hard-code this output?  

```
nc <- 4
n <- 1
test_array <- array(data = rnorm(1000*nc*n), dim = c(1000,nc,n))
y <- posterior::as_draws_array(test_array)
dimnames(y)$variable[1] <- ".log_weight"
posterior::variables(y)
posterior::summarise_draws(y)
```

Reply:
I think it would be desireble to hard-code it, yes, or to throw an error even? Not sure how a consistent output would look like even.

Reply:
I've hard-coded with a warning just in case the current behavior is actually necessary for somebody inside a giant for-loop where adding the necessary `tryCatch` would be a pain